astrocut/cube_cut.py

```diff
@@ -457,17 +463,102 @@ def _update_primary_header(self, primary_header):
+        tassign = 'SPACECRAFT' if product == 'SPOC' else 'N/A'
+        primary_header['TIMEREF'] = (timeref, 'barycentric correction applied to times')        
+        primary_header['TASSIGN'] = (tassign, 'where time is assigned')
+
+        timesys = 'TDB' if product == 'SPOC' else 'TDT'
+        timesys_desc = 'Barycentric' if product == 'SPOC' else 'Terrestrial'
+        primary_header['TIMESYS'] = (f'{timesys}', f'time system is {timesys_desc} Dynamical Time ({timesys})')
+
         primary_header['BJDREFI'] = (2457000, 'integer part of BTJD reference date')           
         primary_header['BJDREFF'] = (0.00000000, 'fraction of the day in BTJD reference date')    
         primary_header['TIMEUNIT'] = ('d', 'time unit for TIME, TSTART and TSTOP')
 
+        if product == 'TICA':
+
+            # Adding some missing kwds not in TICA (but in Ames-produced SPOC ffis)
+            primary_header['EXTVER'] = ('1', 'extension version number (not format version)')
+            primary_header['SIMDATA'] = ('F', 'file is based on simulated data')
+            primary_header['NEXTEND'] = ('2', 'number of standard extensions')
+            primary_header['TSTART'] = (primary_header['STARTTJD'], 'observation start time in TJD of first FFI')
+            primary_header['TSTOP'] = (primary_header['ENDTJD'], 'observation stop time in TJD of last FFI')
+            primary_header['CAMERA'] = (primary_header['CAMNUM'], 'Camera number')
+            primary_header['CCD'] = (primary_header['CCDNUM'], 'CCD chip number')
+            primary_header['ASTATE'] = ('N/A', 'archive state F indicates single orbit processi')
+            primary_header['CRMITEN'] = (primary_header['CRM'], 'spacecraft cosmic ray mitigation enabled')
+            primary_header['CRBLKSZ'] = ('N/A', '[exposures] s/c cosmic ray mitigation block siz')
+            primary_header['FFIINDEX'] = (primary_header['CADENCE'], 'number of FFI cadence interval of first FFI')
+            primary_header['DATA_REL'] = ('N/A', 'data release version number')
+
+            date_obs = Time(primary_header['TSTART']+primary_header['BJDREFI'], format='jd').iso
+            date_end = Time(primary_header['TSTOP']+primary_header['BJDREFI'], format='jd').iso
+            primary_header['DATE-OBS'] = (date_obs, 'TSTART as UTC calendar date of first FFI')
+            primary_header['DATE-END'] = (date_end, 'TSTOP as UTC calendar date of last FFI')
+
+            primary_header['FILEVER'] = ('N/A', 'file format version')
+            primary_header['RADESYS'] = ('N/A', 'reference frame of celestial coordinates')
+            primary_header['SCCONFIG'] = ('N/A', 'spacecraft configuration ID')
+            primary_header['TIMVERSN'] = ('N/A', 'OGIP memo number for file format')
+
+            # Bulk removal with wildcards. Most of these should only live in EXT 1 header.
+            del primary_header['SC_*']  # removes predicted RA, Dec, Roll, etc
```

Comment:
These header deletions only apply to TICA?

Reply:
Correct, I'm deleting only the TICA header keywords that clutter the headers/are not supposed to be in the primary header/are header keywords that describe a single FFI which could be misleading to users, because the cutouts are stacks of FFIs.